django/forms/fields.py

```diff
@@ -204,6 +422,18 @@ def has_changed(self, initial, data):
         data_value = data if data is not None else ''
         return initial_value != data_value
 
+    def bind_to_form(self, form, name):
+        """
+        Bind this field to a form with given name.
+        Returns an object that will eventually be used by a template when accessing
+        form fields.
+
+        For most fields this will just return a BoundField instance but can be
+        customized to return objects that behave differently or offer additional
+        functionality.
+        """
+        return BoundField(form, self, name)
```

Comment:
Do you envision more complex behavior than returning a custom class? If not, I wonder if `get_bound_field_class()` would be simpler.